### ISSUE TITLE
Zooming should maintain the same center point

### ISSUE THREAD

**Issue by babeuloula** [literallycanvas/literallycanvas#341]
Hello,

When I put a picture in the background, and I'm moving in, the zoom is not done from the center of my image.
Any ideas?

Thank you

**Comment by babeuloula** [comment]
The bug is near to this, I think:

```javascript
var newSize, oldSize;
oldSize = viewportSize * oldScale;
newSize = viewportSize * newScale;

return val + ((oldSize - newSize) / 2);
```

**Comment by irskep** [comment]
Please post a demo. It's hard to determine what you're talking about. Also, when pointing to a line of code, please use line numbers.

**Comment by babeuloula** [comment]
Ok sorry, line 1777 and the code is :

```js
var lc = null;

var backgroundImage = new Image();
backgroundImage.src = 'google_maps.jpg';

lc = LC.init(document.getElementById('map'), {
    imageURLPrefix: 'lib/literallycanvas/images/',
    tools: [
        LC.tools.Pencil,
        LC.tools.Eraser,
        LC.tools.Line,
        LC.tools.Rectangle,
        LC.tools.Ellipse,
        LC.tools.Text,
        LC.tools.Polygon,
        LC.tools.Pan
    ],
    imageSize: {
        width: 2448,
        height: 3241
    },
    backgroundShapes: [
        LC.createShape(
            'Image', {
                x: 0,
                y: 0,
                image: backgroundImage,
                scale: 1
            }
        )
    ],
    secondaryColor: 'transparent',
    backgroundColor: '#FFFFFF'
});
```

![google_maps.jpg](https://cloud.githubusercontent.com/assets/4849233/11310660/4d115794-8fcb-11e5-93ea-6b8fc0cb84cc.jpg)

Try to move the map and zoom on the map. You will see, the zoom doesn't zoom on the center of the viewport.

**Comment by babeuloula** [comment]
did you find something ?

**Comment by irskep** [comment]
Haven't yet had a chance to look, sorry

**Comment by babeuloula** [comment]
Hello, 
Some news ?
Thanks

**Comment by irskep** [comment]
Still haven't looked at it. This is a free time project for me.

You are always free to dig into the source code. I know this is response is a trope in the OSS community but there is honestly no guarantee of timeliness for this project.

**Comment by irskep** [comment]
I just spend a couple hours looking into this and got nowhere.

The truth is that the internal coordinate system is crap, and that is my fault. #322 would fix this, and it's slated for the next major release (0.5). That probably won't come for a few months assuming I keep up the work-on-it-in-my-free-time schedule.

**Comment by babeuloula** [comment]
Hello,
Thank you for watching this.

**Comment by irskep** [comment]
Fixed by #421.
